geopandas/geodataframe.py

```diff
@@ -1635,14 +1635,15 @@ def explode(self, column=None, ignore_index=False, add_multiindex=None, **kwargs
                 return super().explode(column, **kwargs)
 
         if add_multiindex is None:
-            warnings.warn(
-                "Currently, add_multiindex defaults to True, but in the future, "
-                "it will default to False to be consistent with Pandas. "
-                "Use `add_multiindex=True` to keep the current behavior and True/False "
-                "to silence the warning.",
-                FutureWarning,
-                stacklevel=2,
-            )
+            if not ignore_index:
```

Comment:
Can we do it here the same way you did it in geoseries? i.e. 

```py
if add_multiindex is None and not ignore_index:
```

Reply:
The reason that I did this differently here is because `index_parts` (formerly `add_multiindex`) is still used later on, so I wanted it to set to True or False instead of leaving it as None (while in the geoseries case, `index_parts` is only reached after `ignore_index` in the if/elif/else statements, so never reached with `ignore_index=True`. 

Now, since `if index_parts:` also works with `index_parts` being, this maybe didn't really matter ..